src/components/TodoPage/TodoPage.tsx

```diff
@@ -0,0 +1,347 @@
+          currentPosts.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+      })
+      .finally(() => {
+        setTimeout(() => {
+          setDeletingTodoId(null);
+        }, 500);
+      });
+  }
+
+  const handleClearCompleted = async () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+    const completedIds = completedTodos.map(todo => todo.id);
+
+    if (completedIds.length === 0) {
+      return;
+    }
+
+    setDeletingAllTodo(completedIds);
+    setIsLoading(true);
+
+    const errors: number[] = [];
+
+    for (const todo of completedTodos) {
+      try {
+        await deletePost(todo.id);
+      } catch {
+        errors.push(todo.id);
+      }
+    }
+
+    if (errors.length > 0) {
+      setErrorMessage('Unable to delete a todo');
+    }
+
+    setTodos(currentTodos =>
+      currentTodos.filter(todo => !todo.completed || errors.includes(todo.id)),
+    );
+
+    setTimeout(() => {
+      setDeletingAllTodo(null);
+      setIsLoading(false);
+    }, 500);
+  };
+
+  function toggleTodo(todoToUpdate: Todo) {
+    setErrorMessage('');
+
+    setLoadingTodoId(todoToUpdate.id);
+
+    const updatedTodo = {
+      ...todoToUpdate,
+      completed: !todoToUpdate.completed,
+    };
+
+    return updatePost(updatedTodo)
+      .then(newTodo => {
+        setTodos(currentTodo => {
+          return currentTodo.map(tod =>
+            tod.id === newTodo.id ? newTodo : tod,
+          );
+        });
+      })
+      .catch(() => {
+        setErrorMessage('Unable to update a todo');
+      })
+      .finally(() => {
+        setTimeout(() => {
+          setLoadingTodoId(null);
+        }, 500);
+      });
+  }
+
+  function toggleAllTodos() {
+    setErrorMessage('');
+
+    const shouldCompleteAll = !todos.every(tod => tod.completed);
+
+    const todosToUpdate = todos
+      .filter(tod => tod.completed !== shouldCompleteAll)
+      .map(tod => ({
+        ...tod,
+        completed: shouldCompleteAll,
+      }));
+
+    const loadingIds = todosToUpdate.map(tod => tod.id);
+
+    setLoadingAllTodo(loadingIds);
+
+    Promise.all(todosToUpdate.map(tod => updatePost(tod)))
+      .then(newTodos => {
+        setTodos(currentTodos =>
+          currentTodos.map(
+            tod => newTodos.find(updated => updated.id === tod.id) || tod,
+          ),
+        );
+      })
+      .catch(() => {
+        setErrorMessage('Unable to update a todo');
+      })
+      .finally(() => {
+        setTimeout(() => {
+          setLoadingAllTodo(null);
+        }, 500);
+      });
+  }
+
+  const handleEditSubmit = async (
+    id: number,
+    oldTitle: string,
+    newTitle: string,
+  ) => {
+    const trimmedTitle = newTitle.trim();
+
+    if (trimmedTitle === '') {
+      try {
+        await deleteTodo(id);
+      } catch {
+        setErrorMessage('Unable to update a todo');
+
+        throw new Error('Unable to update a todo');
```

Comment:
Do not throw error again (fix everywhere)

Reply:
```suggestion
       setErrorMessage('Unable to update a todo');
```
it should fix current failed test
<img width="627" height="178" alt="Screenshot 2025-07-27 at 17 29 59" src="https://github.com/user-attachments/assets/7b875945-0804-4ab9-a43c-f63e93551807" />